Source/ProgressionSystemRuntime/Private/Data/PSSaveGameData.cpp

```diff
@@ -119,13 +122,39 @@ void UPSSaveGameData::UnlockAllLevels()
 // Retrieves the progression reward based on the end game state for the current level.
 float UPSSaveGameData::GetProgressionReward(EEndGameState EndGameState)
 {
-	constexpr float DefaultMultiplier = 1.0f;
-	const float DifficultyMultiplier = UPSWorldSubsystem::Get().GetDifficultyMultiplier();
-	const FPSRowData& CurrentProgressionSettingsRowData = UPSWorldSubsystem::Get().GetCurrentProgressionSettingsRowByName();
+	constexpr float ProgressionReward = 0.f;
 
-	const float* LevelReward = CurrentProgressionSettingsRowData.ProgressionEndGameValues.Find(EndGameState);
-	const float ProgressionReward = LevelReward ? *LevelReward : DefaultMultiplier;
-	return ProgressionReward * DifficultyMultiplier;
+	const UCurveTable* DifficultyCurveTable = UPSDataAsset::Get().GetProgressionDifficultyMultiplierCurveTable();
+	if (!ensureMsgf(DifficultyCurveTable, TEXT("ASSERT: [%i] %s:\n'DifficultyCurveTable' is not valid!"), __LINE__, *FString(__FUNCTION__)))
+	{
+		return ProgressionReward;
+	}
+
+	static const UEnum* Enum = StaticEnum<EEndGameState>();
+	const FString ContextString = Enum->GetNameStringByValue(static_cast<int64>(EndGameState));
+	FName RowName = FName(*ContextString);
+
+	FCurveTableRowHandle Handle;
+	Handle.CurveTable = DifficultyCurveTable;
+	Handle.RowName = RowName;
+
+	const FRealCurve* Curve = Handle.CurveTable->FindCurve(RowName, ContextString);
+	if (!Curve)
+	{
+		return ProgressionReward;
+	}
+
+	float DifficultyType = static_cast<float>(UGameDifficultySubsystem::Get().GetDifficultyType());
```

Comment:
you say next:
> Difficulty values should be: 1, 2, 4, 8

It's because you read enum type calling `.GetDifficultyType()`, which is bitflags. Instead, call `.GetDifficultyLevel()`, where 0 is easy, 1 is normal, 2 is hard

Reply:
done: https://github.com/moinm3uw/ProgressionSystem/pull/57/commits/22e42b7e4f187d1f23be1a49266d640b39696545